src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -78,51 +100,41 @@ func (client *Client) targetQOS(req *http.Request) int {
 // pipeData pipes data between the client and proxy connections.  It's also
 // responsible for responding to the initial CONNECT request with a 200 OK.
 func pipeData(clientConn net.Conn, connOut net.Conn, req *http.Request) {
-	// Start piping to proxy
+	// Start piping from client to proxy
 	go io.Copy(connOut, clientConn)
 
 	// Respond OK
-	err := respondOK(clientConn, req)
-	if err != nil {
+	if err := respondOK(clientConn, req); err != nil {
 		log.Errorf("Unable to respond OK: %s", err)
 		return
 	}
 
-	// Then start coyping from out to client
+	// Then start coyping from proxy to client
 	io.Copy(clientConn, connOut)
 }
 
 func respondOK(writer io.Writer, req *http.Request) error {
 	defer req.Body.Close()
+
 	resp := &http.Response{
-		StatusCode: 200,
+		StatusCode: http.StatusOK,
 		ProtoMajor: 1,
 		ProtoMinor: 1,
 	}
+
 	return resp.Write(writer)
 }
 
-func respondBadGateway(w io.Writer, msg string) error {
+func respondBadGateway(w io.Writer, msg string) (err error) {
 	log.Debugf("Responding BadGateway: %v", msg)
 	resp := &http.Response{
-		StatusCode: 502,
+		StatusCode: http.StatusBadGateway,
 		ProtoMajor: 1,
 		ProtoMinor: 1,
 	}
-	err := resp.Write(w)
+	err = resp.Write(w)
 	if err == nil {
 		_, err = w.Write([]byte(msg))
 	}
 	return err
 }
-
-// hostIncludingPort extracts the host:port from a request.  It fills in a
-// a default port if none was found in the request.
-func hostIncludingPort(req *http.Request, defaultPort int) string {
```

Comment:
We don't really need this as a separate function. The intention of this code is more evident if we move it to the context in which it's actually being used.